internal/rsl/rsl_test.go

```diff
@@ -709,6 +712,109 @@ func TestGetFirstReferenceEntryForRef(t *testing.T) {
 	assert.Equal(t, 5, len(annotations))
 	assertAnnotationsReferToEntry(t, firstEntry, annotations)
 }
+func TestSkipAllInvalidReferenceEntries(t *testing.T) {
+	t.Run("skip latest entry", func(t *testing.T) {
+		repo, err := git.Init(memory.NewStorage(), memfs.New())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := InitializeNamespace(repo); err != nil {
+			t.Fatal(err)
+		}
+
+		ref := plumbing.NewHashReference(plumbing.ReferenceName("refs/heads/main"), plumbing.ZeroHash)
+
+		if err := repo.Storer.SetReference(ref); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := NewReferenceEntry("refs/heads/main", ref.Hash()).Commit(repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		entry, err := GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		referenceEntry, ok := entry.(*ReferenceEntry)
+
+		if !ok {
+			t.Fatal("invalid entry type")
+		}
+
+		assert.Equal(t, referenceEntry.TargetID, plumbing.ZeroHash)
+
+		if err := SkipAllInvalidReferenceEntries(repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		nextEntry, err := GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		annotationEntry, ok := nextEntry.(*AnnotationEntry)
+
+		if !ok {
+			t.Fatal("invalid entry type")
+		}
+
+		assert.Equal(t, annotationEntry.RSLEntryIDs, []plumbing.Hash{entry.GetID()})
+	})
+	t.Run("everything is valid, nothing should change", func(t *testing.T) {
+		repo, err := git.Init(memory.NewStorage(), memfs.New())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := InitializeNamespace(repo); err != nil {
+			t.Fatal(err)
+		}
+
+		commitID, err := gitinterface.CommitUsingSpecificKey(repo, plumbing.ZeroHash, "refs/heads/main", "", gpgKeyBytes)
```

Comment:
do we need a signature? The conditions we check at this package should not worry about the signature at all, no?

Reply:
I have removed the commit using specific key. Thanks